dns/dns.go

```diff
@@ -20,13 +20,15 @@ const (
 	numBuckets   = 15
 )
 
+// Config provides the necessary configuration for creating a Collector.
 type Config struct {
 	Logger micrologger.Logger
 
 	Hosts []string
 }
 
-type DNSCollector struct {
+// Collector implements the Collector interface, exposing DNS latency information.
+type Collector struct {
```

Comment:
main thing here is removing the stuttering. The rest is either alignment with that, or comments.